oci/spec_opts_linux.go

```diff
@@ -48,11 +48,12 @@ func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
 		}
 		for _, dev := range devs {
 			s.Linux.Devices = append(s.Linux.Devices, dev)
+			devNew := dev
```

Comment:
Do we want this to be above the line above? Currently it reads as if we only use the new variable for the code below

Reply:
The code works both way so only readability of the code will be changed. The way it is now shows that new variable  is needed  for the following assignment (since an address is required). A comment should be added?

Reply:
Maybe this way would solve the problem:
```
diff --git a/oci/spec_opts_linux.go b/oci/spec_opts_linux.go
index 1f9afafe3..c840579e9 100644
--- a/oci/spec_opts_linux.go
+++ b/oci/spec_opts_linux.go
@@ -46,14 +46,13 @@ func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
 		if err != nil {
 			return err
 		}
-		for _, dev := range devs {
+		for i, dev := range devs {
 			s.Linux.Devices = append(s.Linux.Devices, dev)
-			devNew := dev
 			s.Linux.Resources.Devices = append(s.Linux.Resources.Devices, specs.LinuxDeviceCgroup{
 				Allow:  true,
-				Type:   devNew.Type,
-				Major:  &devNew.Major,
-				Minor:  &devNew.Minor,
+				Type:   devs[i].Type,
+				Major:  &devs[i].Major,
+				Minor:  &devs[i].Minor,
 				Access: permissions,
 			})
 		}
```

Reply:
Changed to:
```
diff --git a/oci/spec_opts_linux.go b/oci/spec_opts_linux.go
index b16383c5d..ecf7b022a 100644
--- a/oci/spec_opts_linux.go
+++ b/oci/spec_opts_linux.go
@@ -46,13 +46,13 @@ func WithDevices(devicePath, containerPath, permissions string) SpecOpts {
                if err != nil {
                        return err
                }
-               for _, dev := range devs {
-                       s.Linux.Devices = append(s.Linux.Devices, dev)
+               for i := range devs {
+                       s.Linux.Devices = append(s.Linux.Devices, devs[i])
                        s.Linux.Resources.Devices = append(s.Linux.Resources.Devices, specs.LinuxDeviceCgroup{
                                Allow:  true,
-                               Type:   dev.Type,
-                               Major:  &dev.Major,
-                               Minor:  &dev.Minor,
+                               Type:   devs[i].Type,
+                               Major:  &devs[i].Major,
+                               Minor:  &devs[i].Minor,
                                Access: permissions,
                        })
                }
```